### PR TITLE
Prep for Spring, 2019

### LINKED ISSUE #14: Review client-side instructions in LABTASKS.md
There's some basic stuff there right now, and maybe that's all we want. Maybe not.

We need some proofreading here, and perhaps someone has some ideas for how to improve it.
(I do think there's plenty of room for improvement here).

### LINKED ISSUE #27: Clean up the structure of the questions
The question structure was updated right before the lab, and a couple questions were missed in the process. We need to make sure that the structure of the questions is clear and that all of the questions are handled in the same way (e.g., there are not two floating without numbers or links to the questions section). Also, if the questions are not at the bottom, update the text to indicate that (or, move the questions to the bottom?).

### LINKED ISSUE #29: Update the questions to add a definition of "purpose"
The answers we get from students seem to point at what something does, but not why that something matters. We want them to understand what kind of answer and thought we are looking for. We should guide them a bit more toward that.

### LINKED ISSUE #31: Alter LABTASKS.md to make stories that "slice the cake"
It would make sense to ask students to update the API and then the corresponding front end piece so that the ZenHub/story experience is more like it will be in the project. This would give more opportunities to switch between driving and navigating. It also gives more opportunity to switch between front and and back end parts of the system and notice how they connect. We want to provide _stories_. This became clear to us as we were grading and came across this commit message: https://github.com/UMM-CSci-3601-F17/lab-2-client-server-architecture-with-java-spark-bettah/commit/f5b8cf527be6c7abef9c0481650d2fe156472cc9

### LINKED ISSUE #33: Update README.md to say more about what students will need to do to get started
The main thing that seems to be missing is how to handle the gradle window when students checkout this project from github.

### LINKED ISSUE #39: Clarify how to run the project and the tests
Explain that `runWithTests` is really for Travis and they should do the right-click run tests (possibly with coverage) thing in the project view. We should clarify that they can run all the tests, or just tests for a folder, or even for a single file.

Clarify that `run` runs the server but not any of the tests, and will continue running until explicitly stopped, which you should do before you logout and leave the lab.

### LINKED ISSUE #40: Clarify that answers should be in an `ANSWERS.md` file
I think it may say this in the rubric, but it doesn't make it clear in the write-up.

We probably want to create a stub `ANSWERS.md` file just to simplify things.

### LINKED ISSUE #41: Make active links for question 5 in `LABTASKS.md`
Question 5 (What is the purpose of umm3601.Server class?) has several partial URLs in a bullet list. Many students didn't realize that they should (or even could) go to those URLs if they had their server running. We should add full URLs (e.g., `http://localhost:4567/users`) as active links after the fragment so they are encouraged to click those to see what happens.

We should also remind them that they need to have the server running in order for that to work.

### LINKED ISSUE #42: Clarify how to create epics in Zenhub
We ask them to create two epics in Zenhub, but it's not obvious _how_ one does that. We should tell them that they should create an issue, and then click the "make this an epic" thing-y.

### LINKED ISSUE #48: Switch from `ANSWERS.md` to turning things in via Canvas
We currently have them turning in their answers via a file in the repo (`ANSWERS.md`), which it would really make more sense to have them use Canvas (or some other LMS) for that purpose.

We should update the `LABTASKS.md` part of the write-up accordingly.

### PR DESCRIPTION
**Pull request by NicMcPhee**
There's some fairly substantial revisions to `README.md` and `LABTASKS.md` that address a number of outstanding issues.

The one really major change is a complete re-write of the tasks so that
everything now slices the cake. I have them creating an epic for each
feature, with (probably) two stories per epic (one for the server-side
API code, and one for the client-side support).

Things I tried to improve:

   * Specifying the JDK location in IntelliJ
   * Importing a project into IntelliJ
   * What it means to run the server
   * How to run JUnit tests
   * How to set up the Travis badge
   * Fix some broken links
   * Clarify the questions in `LABTASKS.md`
   * Link to some info on how create Epis in ZenHub

 I removed the material about creating a new Gradle task, as that
 didn't actually work and it certainly wasn't very clear.

 We changed from having them create an `ANSWERS.md` file in their repo
 to creating a GDoc with their answers and turning that in via Canvas.

Closes #14
Closes #27
Closes #29
Closes #31
Closes #33
Closes #39
Closes #40
Closes #41
Closes #42
Closes #48

### CODE REVIEW

File: ANSWERS.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+> You should put your answers to the questions in `LABTASKS.md`
```
Comment (kklamberty) [inline ANSWERS.md]: Should it say, "There are questions in LABTASKS.md, and you should put the answers here in ANSWERS.md"? I first read this as you should put the answers in LABTASKS.md.

File: LABTASKS.md [hunk 2]
```diff
@@ -40,12 +40,12 @@ good to deliberately modify some of the tests and see what
...
 
-:warning: One thing we've discovered as we've learned more about
-Spark is that it's fairly complicated to test controller functions that 
+:warning: It's fairly complicated to test Java Spark 
+controller functions that 
 take a `Request` and `Response` object, update the `Response` and return
 a `JsonObject` (like `UserController.getUser`). So at the moment we
 have solid testing of `User` and `Database`, but no direct testing 
-of `UserController`. We'll report back as we figure more out.
+of `UserController`.
```
Comment (kklamberty) [inline LABTASKS.md]: I like this change. It does not promise anything new in the moment, which is a good thing.

**Comment by kklamberty** [comment]
This is fine without any changes, but I found one thing that might need a little clarification. Looks great!

File: ANSWERS.md (hunk 1 above)
Comment (NicMcPhee) [reply ANSWERS.md]: So all that got completely rewritten when I switched from `ANSWERS.md` to GDoc, and I just deleted the `ANSWERS.md` file.

**Comment by NicMcPhee** [comment]
@kklamberty – Can you look this over again? In particular, I *seriously* changed `LABTASKS.md` to now have things "slice the cake" more. I also changed things to use GDoc instead of `ANSWERS.md`. I've changed a lot in the past few hours, so having another pair of eyes look it over would be a Very Good Thing.

Thanks!

**Comment by kklamberty** [comment]
I really like this new splitting of the lab tasks. Thanks for tackling that!

**Pull request merged**